### ISSUE TITLE
KeePassXC 2.5.3 Export to CSV Fails To Export Added or Cloned Entries

### ISSUE THREAD

**Issue by richardstevenhack** [keepassxreboot/keepassxc#4558]
## Expected Behavior

The menu command "Export to CSV file"  to export the entire database, including all cloned and added entries.

## Current Behavior

When I first created the database many months ago, I added all the desired entries. Recently I decided to keep the database as a "master password file" and create a PDF containing my less-critical passwords in order to not have to open the password database every time I needed a trivial password.

So I exported the database to a CSV file, then imported that CSV file into LibreOffice Calc as a spreadsheet, edited the spreadsheet to remove critical password entries, then exported the spreadsheet as a PDF to which I refer daily.

Today I added a database entry by cloning an existing entry, and edited it to change the password and the description.

I then added a database entry by manually adding an entry for a completely unrelated site and password which did not previously exist in the database.

I then attempted to export the database to a CSV file as I did before. I then imported the CSV file into LibreOffice Calc as before. The two entries I changed and added were not present. They were not present in the resulting spreadsheet, nor were they present when I tried it again and examined the LibreOffice Calc prompt describing the layout of the CSV file before actually importing it to the spreadsheet.

In addition, I then tried to use the keepassxc-cli command line to export the database an XML file to see if that would behave differently. First I noticed that although the help prompt from "keepassxc-cli help" shows "export" as an option, when I attempted to use the export command in *interactive mode*, it reports that export is an invalid option. When I select the help option from the interactive mode, it does *not* show "export" as an option. There appears to be an inconsistency there.

`Passwords> help
Available commands:
add            Add a new entry to a database.
analyze        Analyze passwords for weaknesses and problems.
clip           Copy an entry's password to the clipboard.
close          Close the currently opened database.
create         Create a new database.
diceware       Generate a new random diceware passphrase.
edit           Edit an entry.
estimate       Estimate the entropy of a password.
exit           Exit interactive mode.
generate       Generate a new random password.
help           Display command help.
locate         Find entries quickly.
ls             List database entries.
merge          Merge two databases.
mkdir          Adds a new group to a database.
mv             Moves an entry to a new group.
open           Open a database.
quit           Exit interactive mode.
rm             Remove an entry from the database.
rmdir          Removes a group from a database.
show           Show an entry's information.`

I then attempted to do an export directly from the command line without the interactive mode. As has been previously noted by others bug reports, there was no prompt for a password and while an XML was generated, it was only 107 bytes long and contained only an error message as follows:

This page contains the following errors:
error on line 1 at column 1: Document is empty
Below is a rendering of the page up to the first error.

## Possible Solution

None.

## Steps to Reproduce
1. Clone a password entry.
2. Add a password entry manually.
3. Execute the menu command "Export to CSV file".

## Context

This removes my ability to create a subset of the database containing only my non-critical passwords which is up to date.

## Debug Info
KeePassXC - Version 2.5.3
Revision: f8c962b

Qt 5.9.7
Debugging mode is disabled.

Operating system: openSUSE Leap 15.1
CPU architecture: x86_64
Kernel: linux 4.12.14-lp151.28.44-default

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.2

**Comment by droidmonkey** [comment]
I am having a hard time getting past your workflow. This doesn't make any sense to me at all, I'm sorry.

As for the CSV export, there is no reason why new entries wouldn't be exported. We don't do anything particularly special there.

Export from interactive mode is specifically disabled for technical reasons.

**Comment by phoerious** [comment]
What keeps you from creating a database with an empty password? That is definitely better and more convenient than a PDF for all intents and purposes.

**Comment by richardstevenhack** [comment]
Let me point out that using KeePassXC at all requires logging in every time you use it. For trivial passwords, I just want a document I can pop up without any logging in.

What part of "trivial passwords" don't you understand?

As for creating another database, empty password or not, I was specifically trying to avoid using *any* database for the trivial passwords, let alone going through the trouble of creating one and having to import or worse, manually add, passwords I *already have* in the main database.

This is not responsive to *why* export is *not working.* I am not required to explain my workflow to you. Either produce a useful response or allow someone else to do so.

**Comment by droidmonkey** [comment]
I replicated your steps (changed 1 password, added 1 new entry) in between exports. I could not reproduce your problem.

test1.csv:
```
"Group","Title","Username","Password","URL","Notes"
"Root","test","","","",""
"Root","test ","","","",""
"Root","test 2","","","google.com",""
"Root","test","","","usaa.com",""
"Root","test","","","aa.com",""
"Root","new entry 2","","","bing.com",""
```

test2.csv
```
"Group","Title","Username","Password","URL","Notes"
"Root","test","","","",""
"Root","test ","","","",""
"Root","test 2","","","google.com",""
"Root","test","","","usaa.com",""
"Root","test","","","aa.com",""
"Root","new entry 2","","e4zf5ukeh5x3kqscc9ewkhyhp","bing.com",""
"Root","new new entry","","","",""
```

**Comment by richardstevenhack** [comment]
I will repeat the effort right now, as another test, since I have the database open at the moment. I will skip the clone part and just manually add an entry to see if the cloning was at fault in the previous case.

OK, I added a new entry titled "Test" with the same sort of username I use for many sites and a password I use for trivial sites which consists of one word. Saved. Exported it to a CSV file in my usual work directory which contains no other CSV files. I clicked on the CSV file which started LibreOffice which popped up the Text Import dialog with a preview of the CSV file. The added entry was not there.

Ah! I just opened the CSV file with a *text editor*. The added entry is there! The previously cloned entry and manually added entry from the other day is there as well!

However, what I find interesting is the the entries are added to the bottom of the CSV file. I assume this means that updating the database stores the entries at the bottom of whatever database structure you're using.

Went ahead and did the import into a newly created spreadsheet. All the entries are there. They are just not in the alphabetical order all the other entries are. They're on the bottom of the spreadsheet. I was assuming the entries would be in the same order as they appear in the database in KeePassXC.

So I no longer have a real problem. I just have to move the entries to their proper place in the spreadsheet.

You might consider modifying the Database Export command to arrange the CSV in the same order as the database entries appear when viewing the database.

Thank you. Problem solved on my end.
